src/components/common/Stars/Stars.js

```diff
@@ -0,0 +1,147 @@
+import React, { useState } from 'react';
+import styles from './Stars.module.scss';
+import PropTypes from 'prop-types';
+import '../../../styles/settings.scss';
+import { FaStar, FaRegStar } from 'react-icons/fa';
+
+function Stars(props) {
+  const defaultStarsData = [
+    { id: 1, hover: false },
+    { id: 2, hover: false },
+    { id: 3, hover: false },
+    { id: 4, hover: false },
+    { id: 5, hover: false },
+  ];
+
+  const [data, setData] = useState(defaultStarsData);
+
+  const [touch, setTouch] = useState(false);
+
+  function changeStars(id) {
+    const newStarsData = [...defaultStarsData];
+    const exactsIndexs = defaultStarsData.filter(element => element.id <= id);
+    const arrayWithExactsIndexs = [];
+    exactsIndexs.filter(element => arrayWithExactsIndexs.push(element.id));
+    for (let i of newStarsData) {
+      if (arrayWithExactsIndexs.includes(i.id)) {
+        newStarsData[i.id - 1] = {
+          ...i,
+          hover: true,
+        };
+      }
+    }
+    setData(newStarsData);
+  }
+
+  function returnToBasicForm() {
+    setData(defaultStarsData);
+  }
+
+  function touched() {
+    setTouch(true);
+  }
+
+  function noTouched() {
+    setTouch(false);
+  }
+
+  return (
+    <div
+      className={styles.stars}
+      onMouseEnter={() => touched()}
+      onMouseLeave={() => noTouched()}
+      onDoubleClick={e => (
+        props.delateMyStarsChoice(props.stars, props.id), e.preventDefault()
+      )}
+    >
+      {data.map(i => (
+        <a key={i.id} href='#'>
+          {props.myStarsChoice !== 0 ? (
+            touch ? (
+              i.hover ? (
+                <FaStar
```

Comment:
Bardzo dużo tutaj kodu i dużo się powtarza. Spróbuj to wydzielić do osobnych funkcja które zwrócą JSX. Np. `getStar` a w niej skorzystaj z innych pomocniczych funkcji np. `getStarClassName`, i te funkcje powinno przyjmować to od czego zależą, czyli np. `isHover` `isTouch`